Reference app get crashed when it get launched.

**Describe the bug**
Reference app get crashed when it get launched.

**Component**
 reference app

**To Reproduce**
Steps to reproduce the behavior:
1. Just launch the reference app.

**Expected behavior**
Reference app should not be crashed when it get launched.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Smartphone (please complete the following information):**
 - Device: [Moto G5 Plus]
 - Android version: [8.1]

**Additional context**

Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.content.Context.getSystemService(java.lang.String)' on a null object reference
        at android.content.ContextWrapper.getSystemService(ContextWrapper.java:803)
        at androidx.room.RoomDatabase$JournalMode.resolve(RoomDatabase.java:663)
        at androidx.room.RoomDatabase$Builder.build(RoomDatabase.java:1346)
        at com.google.android.fhir.db.impl.DatabaseImpl.<init>(DatabaseImpl.kt:59)
        at com.google.android.fhir.FhirServices$Builder.build(FhirServices.kt:40)
        at com.google.android.fhir.FhirEngineBuilder.build(FhirEngineBuilder.kt:26)
        at com.google.android.fhir.reference.FhirApplication.constructFhirEngine(FhirApplication.kt:47)
        at com.google.android.fhir.reference.FhirApplication.<init>(FhirApplication.kt:32)
        at java.lang.Class.newInstance(Native Method)

**Would you like to work on the issue?**
Please state if this issue should be assigned to you or who you think could help to solve this issue.

--- a/reference/src/main/java/com/google/android/fhir/reference/FhirApplication.kt
+++ b/reference/src/main/java/com/google/android/fhir/reference/FhirApplication.kt
@@ -29,7 +29,11 @@ import kotlinx.coroutines.launch
 import org.hl7.fhir.r4.model.ResourceType
 
 class FhirApplication : Application() {
-  init {
+  // Only initiate the FhirEngine when used for the first time, not when the app is created.
+  private val fhirEngine: FhirEngine by lazy { constructFhirEngine() }
+
+  override fun onCreate() {
+    super.onCreate()
     GlobalScope.launch {
       Sync.oneTimeSync(
         fhirEngine,
@@ -38,9 +42,6 @@ class FhirApplication : Application() {
       )
     }
   }
-
-  // only initiate the FhirEngine when used for the first time, not when the app is created
-  private val fhirEngine: FhirEngine by lazy { constructFhirEngine() }
 
   private fun constructFhirEngine(): FhirEngine {
     return FhirEngineBuilder(this).build()

